File: server/src/main/java/freelance/twin/sport/server/product/entity/ProductVars.java
package freelance.twin.sport.server.product.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

import java.time.LocalDateTime;

@Getter
@Setter
@Entity
@Table(name = "product_variants")
public class ProductVars {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "product_id", nullable = false)
    private Product product;

    private String size;

    private String color;

    private int soldQuantity = 0;

    private int availableQuantity = 0;

    private int stock;

}

File: front/src/models/ProductVars.ts
import { Product } from "./Product";

export interface ProductVars {
  id: number;
  product: Product;
  size?: string;
  color?: string;
  stock: number;
  soldQuantity: number;
  availableQuantity: number;
}

export interface ProductVariantDTO {
  id: number;
  productId: string;
  size?: string;
  color?: string;
  stock: number;
  soldQuantity: number;
  availableQuantity: number;
}